listenbrainz/webserver/static/js/src/CBReviewModal.tsx

```diff
@@ -0,0 +1,703 @@
+          You are reviewing
+          <span className="dropdown">
+            <button
+              className="dropdown-toggle btn-transparent"
+              data-toggle="dropdown"
+              type="button"
+            >
+              {`${entityToReview.name} 
+              (${entityToReview.type.replace("_", " ")})`}
+              <span className="caret" />
+            </button>
+
+            <ul className="dropdown-menu" role="menu">
+              {/* Map entity to dropdown option button */}
+              {allEntities.map((entity) => {
+                if (entity) {
+                  return (
+                    <button
+                      key={entity.mbid}
+                      onClick={() => this.setState({ entityToReview: entity })}
+                      type="button"
+                    >
+                      {`${entity.name} (${entity.type.replace("_", " ")})`}
+                    </button>
+                  );
+                }
+                return <></>;
+              })}
+            </ul>
+          </span>
+          for <a href={this.CBBaseUrl}>CritiqueBrainz</a>.{this.CBInfoButton}
+        </div>
+
+        <div className="form-group">
+          <textarea
+            className="form-control"
+            id="review-text"
+            placeholder={`Review length must be at least ${this.minTextLength} characters.`}
+            value={textContent}
+            name="review-text"
+            onChange={this.handleTextInputChange}
+            rows={6}
+            style={{ resize: "vertical" }}
+            spellCheck="false"
+            required
+          />
+        </div>
+        <small style={{ display: "block", textAlign: "right" }}>
+          Words: {countWords(textContent)} / Chars: {textContent.length}
+        </small>
+
+        <div className="rating-container">
+          <b>Rating: </b>
+          <Rating
+            className="rating-stars"
+            onClick={(rate: number) => this.setState({ rating: rate })}
+            ratingValue={rating}
+            transition
+            size={20}
+            stars={5}
+          />
+        </div>
+
+        <div className="dropdown">
+          <b>Language: </b>
+          <select
+            id="language-selector"
+            value={language}
+            name="language"
+            onChange={this.handleInputChange}
+          >
+            {this.allLanguagesKeyValue.map((lang: any) => {
+              return (
+                <option key={lang[0]} value={lang[0]}>
+                  {lang[1]}
+                </option>
+              );
+            })}
+          </select>
+        </div>
+
+        <div className="checkbox">
+          <label>
+            <input
+              id="acceptLicense"
+              type="checkbox"
+              checked={acceptLicense}
+              name="acceptLicense"
+              onChange={this.handleInputChange}
+              required
+            />
+            <small>
```

Comment:
Damn, that's a long and intimidating checkbox…
I guess there's no good way around it, just like in CB itself.

Reply:
Agreed.. I was wondering if we could move it to a popup window or it's own page instead, but this is dangerous since people might miss it entirely.